### ISSUE TITLE
`lines-around-comment` - `allowBlockStart` doesn't apply to interfaces and types

### ISSUE THREAD

**Issue by GreenTea124** [eslint/eslint#17095]
### Environment

Node version: v16.14.0
npm version: ^8.36.0
Local ESLint version: 
Global ESLint version: (none)
Operating System: Windows 10

### What parser are you using?

@typescript-eslint/parser

### What did you do?

Added this rule:
```
"lines-around-comment": [
	"error",
	{
		allowClassStart: true,
		allowObjectStart: true,
		allowArrayStart: true,
		allowBlockStart: true,
		afterHashbangComment: true,
		beforeBlockComment: true,
		beforeLineComment: true
	}
],
```

### What did you expect to happen?

Expected no linting errors here:
```ts
interface ComponentInterface {
	/*
		aaaa
	*/
	className?: string;

	/*
		aaaa
	*/
	children: React.ReactNode;
}

type ComponentType = {
	/*
		aaaa
	*/
	className?: string;

	/*
		aaaa
	*/
	children: React.ReactNode;
}
```

### What actually happened?

Got 2 errors (1st comment in interface, 1st comment in type)

### Link to Minimal Reproducible Example

https://eslint.org/play/#eyJ0ZXh0IjoiLy8gTm90IHN1cmUgaG93IHRvIGVuYWJsZSBUUyBoZXJlXG5cbi8qIGVzbGludCBsaW5lcy1hcm91bmQtY29tbWVudDogW1wiZXJyb3JcIiwge1xuXHRcdGFsbG93Q2xhc3NTdGFydDogdHJ1ZSxcblx0XHRhbGxvd09iamVjdFN0YXJ0OiB0cnVlLFxuXHRcdGFsbG93QXJyYXlTdGFydDogdHJ1ZSxcblx0XHRhbGxvd0Jsb2NrU3RhcnQ6IHRydWUsXG5cdFx0YWZ0ZXJIYXNoYmFuZ0NvbW1lbnQ6IHRydWUsXG5cdFx0YmVmb3JlQmxvY2tDb21tZW50OiB0cnVlLFxuXHRcdGJlZm9yZUxpbmVDb21tZW50OiB0cnVlXG5cdH1cbl0gKi9cbmludGVyZmFjZSBDb21wb25lbnRJbnRlcmZhY2Uge1xuXHQvKlxuXHRcdGFhYWFcblx0Ki9cblx0Y2xhc3NOYW1lPzogc3RyaW5nO1xuXG5cdC8qXG5cdFx0YWFhYVxuXHQqL1xuXHRjaGlsZHJlbjogUmVhY3QuUmVhY3ROb2RlO1xufVxuXG50eXBlIENvbXBvbmVudFR5cGUgPSB7XG5cdC8qXG5cdFx0YWFhYVxuXHQqL1xuXHRjbGFzc05hbWU/OiBzdHJpbmc7XG5cblx0Lypcblx0XHRhYWFhXG5cdCovXG5cdGNoaWxkcmVuOiBSZWFjdC5SZWFjdE5vZGU7XG59XG4iLCJvcHRpb25zIjp7InBhcnNlck9wdGlvbnMiOnsiZWNtYVZlcnNpb24iOiJsYXRlc3QiLCJzb3VyY2VUeXBlIjoibW9kdWxlIiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7ImNvbnN0cnVjdG9yLXN1cGVyIjpbImVycm9yIl0sImZvci1kaXJlY3Rpb24iOlsiZXJyb3IiXSwiZ2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1hc3luYy1wcm9taXNlLWV4ZWN1dG9yIjpbImVycm9yIl0sIm5vLWNhc2UtZGVjbGFyYXRpb25zIjpbImVycm9yIl0sIm5vLWNsYXNzLWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb21wYXJlLW5lZy16ZXJvIjpbImVycm9yIl0sIm5vLWNvbmQtYXNzaWduIjpbImVycm9yIl0sIm5vLWNvbnN0LWFzc2lnbiI6WyJlcnJvciJdLCJuby1jb25zdGFudC1jb25kaXRpb24iOlsiZXJyb3IiXSwibm8tY29udHJvbC1yZWdleCI6WyJlcnJvciJdLCJuby1kZWJ1Z2dlciI6WyJlcnJvciJdLCJuby1kZWxldGUtdmFyIjpbImVycm9yIl0sIm5vLWR1cGUtYXJncyI6WyJlcnJvciJdLCJuby1kdXBlLWNsYXNzLW1lbWJlcnMiOlsiZXJyb3IiXSwibm8tZHVwZS1lbHNlLWlmIjpbImVycm9yIl0sIm5vLWR1cGUta2V5cyI6WyJlcnJvciJdLCJuby1kdXBsaWNhdGUtY2FzZSI6WyJlcnJvciJdLCJuby1lbXB0eSI6WyJlcnJvciJdLCJuby1lbXB0eS1jaGFyYWN0ZXItY2xhc3MiOlsiZXJyb3IiXSwibm8tZW1wdHktcGF0dGVybiI6WyJlcnJvciJdLCJuby1leC1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZXh0cmEtYm9vbGVhbi1jYXN0IjpbImVycm9yIl0sIm5vLWV4dHJhLXNlbWkiOlsiZXJyb3IiXSwibm8tZmFsbHRocm91Z2giOlsiZXJyb3IiXSwibm8tZnVuYy1hc3NpZ24iOlsiZXJyb3IiXSwibm8tZ2xvYmFsLWFzc2lnbiI6WyJlcnJvciJdLCJuby1pbXBvcnQtYXNzaWduIjpbImVycm9yIl0sIm5vLWlubmVyLWRlY2xhcmF0aW9ucyI6WyJlcnJvciJdLCJuby1pbnZhbGlkLXJlZ2V4cCI6WyJlcnJvciJdLCJuby1pcnJlZ3VsYXItd2hpdGVzcGFjZSI6WyJlcnJvciJdLCJuby1sb3NzLW9mLXByZWNpc2lvbiI6WyJlcnJvciJdLCJuby1taXNsZWFkaW5nLWNoYXJhY3Rlci1jbGFzcyI6WyJlcnJvciJdLCJuby1taXhlZC1zcGFjZXMtYW5kLXRhYnMiOlsiZXJyb3IiXSwibm8tbmV3LXN5bWJvbCI6WyJlcnJvciJdLCJuby1ub25vY3RhbC1kZWNpbWFsLWVzY2FwZSI6WyJlcnJvciJdLCJuby1vYmotY2FsbHMiOlsiZXJyb3IiXSwibm8tb2N0YWwiOlsiZXJyb3IiXSwibm8tcHJvdG90eXBlLWJ1aWx0aW5zIjpbImVycm9yIl0sIm5vLXJlZGVjbGFyZSI6WyJlcnJvciJdLCJuby1yZWdleC1zcGFjZXMiOlsiZXJyb3IiXSwibm8tc2VsZi1hc3NpZ24iOlsiZXJyb3IiXSwibm8tc2V0dGVyLXJldHVybiI6WyJlcnJvciJdLCJuby1zaGFkb3ctcmVzdHJpY3RlZC1uYW1lcyI6WyJlcnJvciJdLCJuby1zcGFyc2UtYXJyYXlzIjpbImVycm9yIl0sIm5vLXRoaXMtYmVmb3JlLXN1cGVyIjpbImVycm9yIl0sIm5vLXVuZGVmIjpbImVycm9yIl0sIm5vLXVuZXhwZWN0ZWQtbXVsdGlsaW5lIjpbImVycm9yIl0sIm5vLXVucmVhY2hhYmxlIjpbImVycm9yIl0sIm5vLXVuc2FmZS1maW5hbGx5IjpbImVycm9yIl0sIm5vLXVuc2FmZS1uZWdhdGlvbiI6WyJlcnJvciJdLCJuby11bnNhZmUtb3B0aW9uYWwtY2hhaW5pbmciOlsiZXJyb3IiXSwibm8tdW51c2VkLWxhYmVscyI6WyJlcnJvciJdLCJuby11bnVzZWQtdmFycyI6WyJlcnJvciJdLCJuby11c2VsZXNzLWJhY2tyZWZlcmVuY2UiOlsiZXJyb3IiXSwibm8tdXNlbGVzcy1jYXRjaCI6WyJlcnJvciJdLCJuby11c2VsZXNzLWVzY2FwZSI6WyJlcnJvciJdLCJuby13aXRoIjpbImVycm9yIl0sInJlcXVpcmUteWllbGQiOlsiZXJyb3IiXSwidXNlLWlzbmFuIjpbImVycm9yIl0sInZhbGlkLXR5cGVvZiI6WyJlcnJvciJdfSwiZW52Ijp7ImVzNiI6dHJ1ZX19fQ==

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by mdjermanovic** [comment]
Hi @GreenTea124, thanks for the issue!

ESLint core rules don't support TypeScript-specific syntax.

Instead of the core `lines-around-comment` rule, you can use [`@typescript-eslint/lines-around-comment`](https://typescript-eslint.io/rules/lines-around-comment) rule with options `allowInterfaceStart: true` and `allowTypeStart: true`. Here's a [typescript-eslint demo](https://typescript-eslint.io/play/#ts=5.0.4&sourceType=module&code=PQKgBApgzgNglgOwC5gAJIJ4AdoGMBOcWSAtNPMsBdCQIb4D2ArggCYm4MC2XEyAXGADaAIgj5G+EQBowAbwCwAKDCqwtGDAYB3AJLJxAM1q4IAZST0kgpPiYRpytes06AKtnOX81sLftOYAC+ygC6YCDAyohIRiYQYADC3FgMCHxI+rH4xqbygaCBqrQltIGRgbgwtFBQAHK0vAD8glC2iADmANzKBSBF6qXlUSpguAAWcDCs+HyCAEoQJkgAdIvLdQysED1KIUrKmDhJKWkZHscAvPmjhaPFQ6MVo1U19Y0QLWBthAjdvbd+vdBiVhpVJtNZggFktcKt1nDNttdvsgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA) with your example code.

**Comment by GreenTea124** [comment]
@mdjermanovic Thanks!
